ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/reaper/ReaperTestCase4.java

```diff
@@ -0,0 +1,100 @@
+package com.hp.mwtests.ts.arjuna.reaper;
+
+import com.arjuna.ats.arjuna.common.Uid;
+import com.arjuna.ats.arjuna.coordinator.Reapable;
+import com.arjuna.ats.arjuna.coordinator.TransactionReaper;
+import org.jboss.byteman.contrib.bmunit.BMScript;
+import org.jboss.byteman.contrib.bmunit.BMUnitConfig;
+import org.jboss.byteman.contrib.bmunit.BMUnitRunner;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.Assert.fail;
+
+@RunWith(BMUnitRunner.class)
+@BMScript("reaper")
+@BMUnitConfig(debug = true)
+public class ReaperTestCase4 extends ReaperTestCaseControl {
+
+    private static int _timeout = 5;
+
+    private Reapable reapable = new SimpleReapable();
+
+    private volatile long beforeTerminate;
+    private volatile long afterWait;
+
+    @Test
```

Comment:
@jmfinelli is there a need to keep both of these tests or could it be constrained to a single test again?

Reply:
I was just playing/testing a bit :-) The test without rendezvous is helpful to confirm that the "normal" path (i.e. removeElementClient -> this.notify()) is fired. Of course, the race condition might happen during the test without rendezvous, which will lead to its failure. If we want to keep this additional test, we need to create a byteman rule to force `_inShutdown` to be true